driver-core/src/main/com/mongodb/connection/SaslAuthenticator.java

```diff
@@ -47,6 +47,7 @@ public void authenticate(final InternalConnection connection, final ConnectionDe
             @Override
             public Void run() {
                 SaslClient saslClient = createSaslClient(connection.getDescription().getServerAddress());
+                throwIfSaslClientIsNull(saslClient);
```

Comment:
SaslClient.create is documented to return null rather than throw if no implementation is found for the requested mechanism.